pkg/builder3/openapi.go

```diff
@@ -74,6 +74,7 @@ func (o *openAPI) buildOperations(route common.Route, inPathCommonParamsMap map[
 	ret := &spec3.Operation{
```

Comment:
for openapi v3, what about the deprecated field on the schema object as well as the operation?

Reply:
Yes, that'd be super sweet, especially if we automatically enable new mechanisms for help with deprecation.

Reply:
Yeah it should possible but it's not as straightforward and I thought it might be better to include in a follow up after some more discussion.

I'm still doing some investigations, but afaik we can't guarantee that a deprecated operation path would deprecate all the corresponding schemas associated with that path. An example of this is a delete on an alpha/beta resource that returns a meta.v1.Status. Is this a one-off exception or are there more examples of this?

For schema fields, there are cases where we deprecate a single schema field and not the entire schema. This is currently embedded in field descriptions, and it may be beneficial to include these programmatically as well.

Reply:
If you look at how this deprecated operation setting is being driven from k/k, it's based on the entire primary struct for a set of operations being deprecated. Even if it's in another PR, having a way to plumb the fact that that struct is deprecated into openapi would be useful.

Reply:
OK, thanks for clarifying, I completely agree.